### PR TITLE
TD1320 ImmutableSignedZoneV2 deploy script

### PR DESCRIPTION
**Pull request by frankisawesome**
- Use create3
- Basic test

### CODE REVIEW

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            signer: 0xdDA0d9448Ebe3eA43aFecE5Fa6401F5795c19333,
+            factory: 0x37a59A845Bb6eD2034098af8738fbFFB9D589610,
+            salt: "salty"
+        });
+
+        /// @dev These are Immutable zkEVM testnet values where necessary
+        ZoneDeploymentArgs memory zoneDeploymentArgs = ZoneDeploymentArgs({
+            owner: address(0xC606830D8341bc9F5F5Dd7615E9313d2655B505D),
+            name: "TestImmutableSignedZoneV2",
+            apiEndpoint: "https://api.sandbox.immutable.com/",
```
Comment (naveen-imtb) [inline script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:49]: In the V1 zone deployment script, we've left the `api_endpoint` empty. Having the api endpoint info is fine but the script will need to account for the different values in different evnironments.

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            documentationURI: ""
+        });
+
+        // Run deployment against forked testnet
+       _deploy(deploymentArgs, zoneDeploymentArgs);
+    }
+
+     function deploy() external {
+        address signer = vm.envAddress("DEPLOYER_ADDRESS");
```
Comment (naveen-imtb) [inline script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:58]: What is the role of the signer? 
Does the `msg.sender` have no role in this workflow?

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        /// @dev These are Immutable zkEVM testnet values where necessary
+        DeploymentArgs memory deploymentArgs = DeploymentArgs({
+            signer: 0xdDA0d9448Ebe3eA43aFecE5Fa6401F5795c19333,
+            factory: 0x37a59A845Bb6eD2034098af8738fbFFB9D589610,
+            salt: "salty"
+        });
+
+        /// @dev These are Immutable zkEVM testnet values where necessary
+        ZoneDeploymentArgs memory zoneDeploymentArgs = ZoneDeploymentArgs({
+            owner: address(0xC606830D8341bc9F5F5Dd7615E9313d2655B505D),
```
Comment (naveen-imtb) [inline script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:47]: Should the owner be defaulted to the deployer? Isn't the plan to then transfer ownership of the contract to the multisig wallet?

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol (hunk 1 above)
Comment (frankisawesome) [reply script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:50]: @naveen-imtb this is the test, actual deployments will use env var during deployment

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol (hunk 2 above)
Comment (frankisawesome) [reply script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:70]: The signer is the deployer used

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol (hunk 3 above)
Comment (frankisawesome) [reply script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:48]: Same as above, this is just a test value

**Review by naveen-imtb — APPROVED** [review]

File: script/trading/seaport/DeployImmutableSignedZoneV2.s.sol [hunk 4]
```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: Apache-2
+// solhint-disable-next-line compiler-version
+pragma solidity 0.8.20;
+
+import "forge-std/Test.sol";
+import {ImmutableSignedZoneV2} from "../../../contracts/trading/seaport/zones/immutable-signed-zone/v2/ImmutableSignedZoneV2.sol";
+import "forge-std/console.sol";
```
Comment (lfportal) [inline script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:8]: If I understand correctly, this `import` isn't required if we've already imported `forge-std/Test.sol`

Comment (frankisawesome) [reply script/trading/seaport/DeployImmutableSignedZoneV2.s.sol:8]: I should've removed the console logs all together, will update

Comment (frankisawesome) [reply script/trading/seaport/DeployImmutableSignedZoneV2.s.sol]: [bc5640c](https://github.com/immutable/contracts/pull/215/commits/bc5640c3e5c1748f0e28a3d84a267fd49d7fd2f4)

**Review by lfportal — APPROVED** [review]

**Pull request merged**